internal/oci/container.go

```diff
@@ -707,3 +709,55 @@ func (c *Container) RestoreIsOCIImage() bool {
 func (c *Container) SetRestoreIsOCIImage(restoreIsOCIImage bool) {
 	c.restoreIsOCIImage = restoreIsOCIImage
 }
+
+// SetResources loads the OCI Spec.Linux.Resources in the container struct
+func (c *Container) SetResources(s *specs.Spec) {
+	if s.Linux != nil && s.Linux.Resources != nil {
+		linuxResources := s.Linux.Resources
+		resourceStatus := &types.ContainerResources{}
+		resourceStatus.Linux = &types.LinuxContainerResources{}
+		if linuxResources.CPU != nil {
+			resourceStatus.Linux.CpusetCpus = linuxResources.CPU.Cpus
+			resourceStatus.Linux.CpusetMems = linuxResources.CPU.Mems
+			if linuxResources.CPU.Period != nil {
+				resourceStatus.Linux.CpuPeriod = int64(*linuxResources.CPU.Period)
+			}
+			if linuxResources.CPU.Quota != nil {
+				resourceStatus.Linux.CpuQuota = *linuxResources.CPU.Quota
+			}
+			if linuxResources.CPU.Shares != nil {
+				resourceStatus.Linux.CpuShares = int64(*linuxResources.CPU.Shares)
+			}
+		}
+		if linuxResources.Memory != nil {
+			if linuxResources.Memory.Limit != nil {
+				resourceStatus.Linux.MemoryLimitInBytes = *linuxResources.Memory.Limit
+			}
+			if linuxResources.Memory.Swap != nil {
+				resourceStatus.Linux.MemorySwapLimitInBytes = *linuxResources.Memory.Swap
+			}
+		}
+		if s.Process != nil {
+			if s.Process.OOMScoreAdj != nil {
+				resourceStatus.Linux.OomScoreAdj = int64(*s.Process.OOMScoreAdj)
+			}
+		}
+		resourceStatus.Linux.Unified = make(map[string]string)
+		for key, value := range linuxResources.Unified {
+			resourceStatus.Linux.Unified[key] = value
+		}
+		resourceStatus.Linux.HugepageLimits = []*types.HugepageLimit{}
+		for _, entry := range linuxResources.HugepageLimits {
```

Comment:
Consider adding a unit test for the newly introduced function, it will also make code coverage happy.

Reply:
Done :)